### PR TITLE
asyncBlocking in webRequest.onAuthRequired

### PR DESCRIPTION
**Pull request by rebloor**
### Description

Addresses the documentation requirements of [Bug 1889897](https://bugzilla.mozilla.org/show_bug.cgi?id=1889897) Implement asyncBlocking support for webRequest.onAuthRequired, including:

- a release note 
- update to the [webRequest.onAuthRequired](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/webRequest/onAuthRequired) event documentation. This also included some changes to avoid repetition in that documentation.

### CODE REVIEW

File: files/en-us/mozilla/add-ons/webextensions/api/webrequest/onauthrequired/index.md [hunk 1]
```diff
@@ -83,21 +83,21 @@ Events have three functions:
...
 
   - : `array` of `string`. Extra options for the event. You can pass any of the following values:
 
-    - `"blocking"`: make the request block so you can cancel the request or supply authentication credentials. To handle the request asynchronously in Chrome, use `"asyncBlocking"` instead.
+    - `"blocking"`: make the request block so you can cancel the request or supply authentication credentials. Return a `BlockingResponse` object with its `cancel` or `authCredentials` properties set.
+
+    - `"asyncBlocking"`: handle the request asynchronously.
+
+      - in Firefox up to version 127, `"blocking"` is used instead, and the event handler function can return a Promise that resolves to a `BlockingResponse` object, with its `cancel` or `authCredentials` properties set. This is basically the same as handling the event synchronously.
+      - in Firefox 128 and Chrome, `"asyncBlocking"` is used. The event handler function is passed a function as a second parameter (called `asyncCallback`) that should be invoked with the `BlockingResponse` result, with its `cancel` or `authCredentials` properties set.
```
Comment (dotproto) [inline files/en-us/mozilla/add-ons/webextensions/api/webrequest/onauthrequired/index.md:99]: The return value of listeners registered with `"asyncBlocking"` is ignored. This applies to both Firefox and Chrome. The listener MUST use the callpack passed into the listener via `asyncCallback` to provide a response.

```suggestion
    - `"asyncBlocking"`: handle the request asynchronously. The return value of the event listener is ignored. To resolve the event, pass the `asyncCallback` parameter a `BlockingResponse` result, with its `cancel` or `authCredentials` properties set.
      - Supported in Chrome 120 and later.
      - Supported in Firefox 128 and later.
      - Not supported in Safari.
```

**Review by dotproto — CHANGES_REQUESTED** [review]
It looks like there may have been a miscommunication around how async handling of onAuthRequired events work in Firefox 128+. The version of this PR I'm reviewing seems to say that Firefox 128 and later no longer support asynchronous response handling in `"blocking"` listeners. Based on my reading of Firefox source, it appears that `blocking` listeners can still return promises to asynchronously settle an onAuthRequired events. What has changed is that Firefox now supports registering onAuthRequired listeners with `"asyncBlocking"` and these listeners match the behavior seen in Chrome.

All changes suggested in this review are for illustrative purposes. I'm happy to defer to your judgement on how best to structure the related content.

File: files/en-us/mozilla/add-ons/webextensions/api/webrequest/onauthrequired/index.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ The listener can respond in one of four ways:
 
 - Provide credentials synchronously
 
-  - : If credentials are available synchronously, the extension can supply them synchronously. If the extension does this, the browser attempts to log in with the credentials. The listener can provide credentials synchronously as follows:
+  - : If credentials are available synchronously, the extension can supply them synchronously. However, synchronous handling is only appropriate for the simplest of extensions. If the extension does this, the browser attempts to log in with the credentials. The listener can provide credentials synchronously as follows:
```
[the remark this answers is not in the document]
Comment (rebloor) [reply files/en-us/mozilla/add-ons/webextensions/api/webrequest/onauthrequired/index.md:25]:
```suggestion
  - : If credentials are available synchronously, the extension can supply them synchronously. If the extension does this, the browser attempts to log in with the credentials. The listener can provide credentials synchronously as follows:
```

File: files/en-us/mozilla/add-ons/webextensions/api/webrequest/onauthrequired/index.md [hunk 3]
```diff
@@ -82,23 +84,27 @@ Events have three functions:
...
-  - : {{WebExtAPIRef('webRequest.RequestFilter')}}. A filter that restricts the events that are sent to this listener.
+  - : {{WebExtAPIRef('webRequest.RequestFilter')}}. A filter that restricts the events sent to this listener.
 - `extraInfoSpec` {{optional_inline}}
 
-  - : `array` of `string`. Extra options for the event. You can pass any of the following values:
+  - : `array` of `string`. Extra options for the event. You can pass any of these values:
+
+    - `"blocking"`: make the request block so you can cancel the request or supply authentication credentials. Return a `BlockingResponse` object with its `cancel` or `authCredentials` properties set.
+
+      - In Chrome, the event listener must respond synchronously.
```
Comment (rebloor) [inline files/en-us/mozilla/add-ons/webextensions/api/webrequest/onauthrequired/index.md:102]: @dotproto why if "Chrome only supports "asyncBlocking"."?

**Comment by rebloor** [comment]
@dotproto I need to check the language in the changes and fix some formatting issues but in the meanwhile, could you check my latest updates, do they address your feedback?

**Comment by dotproto** [comment]
@rebloor, I just created a test extension to flex the various ways that an `onAuthRequired` event handler can try to resolve an auth request. Hopefully this helps clear up the confusion about the behavior differences between Chrome and Firefox. https://gist.github.com/dotproto/43cba54752300484b69ca0d0ecc1c82c

**Review by dotproto — CHANGES_REQUESTED** [review]

File: files/en-us/mozilla/add-ons/webextensions/api/webrequest/onauthrequired/index.md [hunk 4]
```diff
@@ -82,23 +84,28 @@ Events have three functions:
 
     - `details`
       - : `object`. Details about the request. See the [details](#details_2) section for more information.
+    - `asyncCallback` {{optional_inline}}
+
```
[the remark this answers is not in the document]
Comment (dotproto) [reply files/en-us/mozilla/add-ons/webextensions/api/webrequest/onauthrequired/index.md]: @rebloor, any idea why mdn-linter is generating linter warnings that aren't showing up when doing a local `yarn lint:md` or know who we could tag about that?

**Review by dotproto — APPROVED** [review]
Just one last subjective suggestion. Thanks, @rebloor!

File: files/en-us/mozilla/add-ons/webextensions/api/webrequest/onauthrequired/index.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (rebloor) [reply files/en-us/mozilla/add-ons/webextensions/api/webrequest/onauthrequired/index.md]: @dotproto sorry, no and no

File: files/en-us/mozilla/add-ons/webextensions/api/webrequest/onauthrequired/index.md [hunk 5]
```diff
@@ -31,8 +31,10 @@ The listener can respond in one of four ways:
 
   - : The extension might need to fetch credentials asynchronously. For example, the extension might need to fetch credentials from storage or ask the user. In this case, the listener can supply credentials asynchronously as follows:
 
-    - in addListener, pass `"blocking"` in the `extraInfoSpec` parameter
-    - in the listener, return a `Promise` that resolves with an object containing an `authCredentials` property, set to the credentials to supply
+    - in addListener, pass `"asyncBlocking"` in Chrome and Firefox or `"blocking"` in Firefox in the `extraInfoSpec` parameter
+    - If `"blocking"` is provided, the extension can return a `webRequest.BlockingResponse` object or a Promise that resolves to a `webRequest.BlockingResponse` object
+    - If `"asyncBlocking"` is provided, the event listener function receives a `asyncCallback` function as its second parameter. `asyncCallback`can be called asynchronously and takes a`webRequest.BlockingResponse` object as its only parameter
+
       > **Note:** Chrome does not support a Promise as a return value ([Chromium issue 1510405](https://crbug.com/1510405)). For alternatives, see [the return value of the `listener`](#listener).
```
Comment (dotproto) [inline files/en-us/mozilla/add-ons/webextensions/api/webrequest/onauthrequired/index.md:38]: It looks like this information is a bit redundant given the recent changes to the `extraInfoSpec` parameter documentation. Should we remove this block and link to that property for details? I'm happy to defer to your judgement.

**Pull request merged**